packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -0,0 +1,102 @@
+<script lang="ts" setup generic="TValue extends SelectionOptionValue = SelectionOptionValue">
+import { createListbox } from "@sit-onyx/headless";
+import { computed } from "vue";
+import OnyxListboxOption from "../OnyxListboxOption/OnyxListboxOption.vue";
+import type { SelectionOptionValue } from "../OnyxRadioButton/types";
+import type { OnyxListboxProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxListboxProps<TValue>>(), {
+  hideLabel: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the current value changes.
+   */
+  "update:modelValue": [value: typeof props.modelValue];
+}>();
+
+const {
+  elements: { listbox, option: headlessOption },
+} = createListbox({
+  label: computed(() => props.label),
+  onSelect: (id) => {
+    if (props.modelValue === id) emit("update:modelValue", undefined);
+    else emit("update:modelValue", id as TValue);
+  },
+});
+</script>
+
+<template>
+  <div class="onyx-listbox">
+    <ul class="onyx-listbox__options" v-bind="listbox">
+      <OnyxListboxOption
+        v-for="option in props.options"
+        :key="option.id.toString()"
+        class="onyx-listbox__option"
+        :headless-option="
+          headlessOption({
+            id: option.id,
+            label: option.label,
+            disabled: option.disabled,
+            selected: option.id === props.modelValue,
+          })
```

Comment:
Can we flatten the API?
Not really a fan of a property called `headlessOption` .

```suggestion
        v-bind="
          headlessOption({
            id: option.id,
            label: option.label,
            disabled: option.disabled,
            selected: option.id === props.modelValue,
          })
```

Reply:
I wanted to do this first but then the Vue compiler crashes because it can not resolve the type...

Reply:
What about defining the Props explicitly or/and using fallthrough props?

Reply:
I used fallthrough attributes :)